src/Elastic.Apm.AspNetCore/ApmMiddlewareExtension.cs

```diff
@@ -52,14 +52,13 @@ public static class ApmMiddlewareExtension
 				? new EnvironmentConfigurationReader(logger)
 				: new MicrosoftExtensionsConfig(configuration, logger, builder.ApplicationServices.GetEnvironmentName()) as IConfigurationReader;
 
-			if (!configReader.Enabled)
-				return builder;
-
 			var config = new AgentComponents(configurationReader: configReader, logger: logger);
 			HostBuilderExtensions.UpdateServiceInformation(config.Service);
 
+			// Agent.Setup must be called, even if agent is disabled. This way static public API usage won't implicitly initialize an agent with default values, instead, this will be reused.
 			Agent.Setup(config);
-			return UseElasticApm(builder, Agent.Instance, logger, subscribers);
+
+			return !configReader.Enabled ? builder : UseElasticApm(builder, Agent.Instance, logger, subscribers);

```

Comment:
can this just call `return UseElasticApm(builder, Agent.Instance, logger, subscribers);` and let `UseElasticApm(...)` return without setting up middleware?

Reply:
Yes 👍  Makes sense.